tests/plugins/emoji/dropdown.js

```diff
@@ -359,6 +359,35 @@
 					}
 				} );
 			} );
+		},
+
+		// (#2607)
+		'test SVG icons support': function() {
+			if ( !this.editor.plugins.emoji.isSVGSupported() ) {
+				assert.ignore();
+			}
+
+			var bot = this.editorBot;
+
+			bot.panel( 'EmojiPanel', function( panel ) {
+				var doc = panel._.iframe.getFrameDocument();
+
+				CKEDITOR.ajax.load( this.editor.plugins.emoji.path + 'assets/iconsall.svg', function( html ) {
+					var container = new CKEDITOR.dom.element( 'div' );
+
+					container.setHtml( html );
+
+					resume( function() {
+						var icons = doc.findOne( '.cke_emoji-navigation_icons' );
+
+						assert.beautified.html( container.getHtml(), icons.getHtml(), 'Icons should be loaded as a part of panel' );
+
+						panel.hide();
+					} );
```

Comment:
This code mocks most of the logic from `loadSVGNavigationIcons()` method which means it doesn't really test it. Wouldn't it be possible to call `loadSVGNavigationIcons()` here or create more integration like test so the ajax call would be performed by the plugin itself?

Reply:
What do you mean about mocking? :thinking: Ajax request in unit test loads SVG HTML for the simpler assertion, so we don't need to copy the same HTML in 2 different places. But `loadSVGNavigationIcons` is called without any mocking by the panel itself once opened.

Reply:
> Ajax request in unit test loads SVG HTML for the simpler assertion, so we don't need to copy the same HTML in 2 different places. But `loadSVGNavigationIcons` is called without any mocking by the panel itself once opened.

Ok, I see now, I somehow missed that, thinking that icons loading is done only by ajax request  :thinking: So this test does exactly what I thought it should :+1: